cli/src/commands/examples.rs

```diff
@@ -0,0 +1,234 @@
+
+        let asset_name = asset.name.clone();
+        let mut split_asset_name = asset_name.splitn(2, '-');
+
+        // First part is language
+        let language = if let Some(lang) = split_asset_name.next() {
+            lang.to_string()
+        } else {
+            // Bad name, just ignore it
+            continue;
+        };
+        let language_display_name = capitalize(&language);
+
+        // Second is example name
+        let example_display_name = if let Some(example) = split_asset_name.next() {
+            example
+                .to_string()
+                .from_case(Case::Kebab)
+                .to_case(Case::Title)
+        } else {
+            // Bad name, just ignore it
+            continue;
+        };
+        let example_display_name = example_display_name.trim_end_matches(".zip").to_string();
+
+        languages_map
+            .entry(language)
+            .or_insert_with(|| Language {
+                display_name: language_display_name,
+                examples: vec![],
+            })
+            .examples
+            .push(Example {
+                display_name: example_display_name,
+                asset,
+            });
+    }
+
+    languages_map.into_values().collect()
+}
+
+fn capitalize(s: &str) -> String {
+    let mut c = s.chars();
+    match c.next() {
+        None => String::new(),
+        Some(f) => f.to_uppercase().collect::<String>() + c.as_str(),
+    }
+}
+
+async fn download_example(repo_handler: RepoHandler<'_>, asset: Asset) -> Result<()> {
+    let out_dir_name = PathBuf::from(asset.name.trim_end_matches(".zip").to_string());
+    // This fails if the directory already exists.
+    tokio::fs::create_dir(&out_dir_name).await.context(format!(
+        "Cannot create the output directory {} for the example",
+        out_dir_name.display()
+    ))?;
+
+    let mut zip_out_file_path = PathBuf::from(&out_dir_name);
+    zip_out_file_path.push("temp.zip");
+
+    // Download zip in the out_dir
+    match download_asset_to_file(&zip_out_file_path, repo_handler, asset).await {
+        Ok(f) => f,
+        Err(e) => {
+            // Try to remove the directory, to avoid leaving a dirty user directory in case of a retry.
+            let _ = tokio::fs::remove_dir_all(&out_dir_name).await;
+            bail!(
+                "Error when downloading the zip {} for the example: {:#?}",
+                zip_out_file_path.display(),
+                e
+            );
+        }
+    };
+
+    // Unzip it
+    if let Err(e) = unzip(&zip_out_file_path, &out_dir_name).await {
+        // Try to remove the directory, to avoid leaving a dirty user directory in case of a retry.
+        println!("{:#?}", e);
+        let _ = tokio::fs::remove_dir_all(&out_dir_name).await;
+        return Err(e);
+    }
+
+    // Ready to rock!
+    c_println!(
+        "The example is ready in the directory {}",
+        out_dir_name.display()
```

Comment:
Could be cool to style the directory name with `Styled(Style::Success, out_dir_name.display())`